boards: nxp: s32k5xxcvb: add support for watchdog

Dat-NguyenDuy opened the pull request:
Add support for Watchdog (SWT)
Tested `wdt_basic_reset_none` for both core
Tested `samples\drivers\watchdog` and `wdt_basic_api` for core m7 only

haduongquang committed:
boards: nxp: s32k5xxcvb: add support for watchdog

Add support for Watchdog (SWT)

Signed-off-by: Ha Duong Quang <ha.duongquang@nxp.com>
Co-authored-by: Dat Nguyen Duy <dat.nguyenduy@nxp.com>
Signed-off-by: Dat Nguyen Duy <dat.nguyenduy@nxp.com>

JarmouniA reviewed (dismissed):
Please split into 4 commits: SoC dts, boards, samples, tests

Dat-NguyenDuy committed:
dts: nxp: add watchdog devicetree nodes for s32k566

Add watchdog devicetree nodes for S32K566

Signed-off-by: Dat Nguyen Duy <dat.nguyenduy@nxp.com>

Dat-NguyenDuy committed:
samples: support watchdog samples for s32k5xxvcb

Only enable watchdog sample for cm7. For cr52, because the
SoC set the cores in Thumb mode after reset, so a debugger
is always needed to switch the core to Arm state before
loading zephyr application.

Signed-off-by: Dat Nguyen Duy <dat.nguyenduy@nxp.com>

Dat-NguyenDuy committed:
tests: drivers: support watchdog tests for s32k5xxcvb

Enable wdt_basic_api watchdog only for cm7. For cr52,
because the SoC set the cores in Thumb mode after reset,
so a debugger is always needed to switch the core to Arm
state before loading zephyr application.

wdt_basic_reset_none test can be enabled for both.

Signed-off-by: Dat Nguyen Duy <dat.nguyenduy@nxp.com>

manuargue commented on samples/drivers/watchdog/sample.yaml:
```diff
@@ -22,6 +22,8 @@ tests:
       - s32z2xxdc2/s32z270/rtu1
       - s32z2xxdc2@D/s32z270/rtu0
       - s32z2xxdc2@D/s32z270/rtu1
+      - s32k5xxcvb/s32k566/m7
```
I'd suggest renaming `sample.drivers.watchdog.s32z270dc2_r52` to `sample.drivers.watchdog.nxp_s32_swt` and adding s32k5 platform to this build-only scenario

Dat-NguyenDuy replied:
I've updated

manuargue commented on samples/drivers/watchdog/sample.yaml:
>samples: exclude watchdog samples for s32k5xxvcb
Only wdt_basic_reset_none test can be executed, the rest
should be excluded

pls update the commit msg, it's referencing a test but the commit is updating a sample

Dat-NguyenDuy replied:
I've updated

Dat-NguyenDuy commented:
Fixed compliances

Dat-NguyenDuy commented:
Fixed conflicts

Dat-NguyenDuy commented:
I've resolved conflicts

Dat-NguyenDuy commented on boards/nxp/s32k5xxcvb/s32k5xxcvb_s32k566.dtsi:
I've updated
